pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -257,9 +271,13 @@ func (f *framework) QueueSortFunc() LessFunc {
 func (f *framework) RunPreFilterPlugins(
 	ctx context.Context, state *CycleState, pod *v1.Pod) (status *Status) {
 	startTime := time.Now()
-	defer func() { recordExtensionPointDuration(startTime, preFilter, status) }()
+	defer func() {
+		f.metricsRecorder.observeExtensionPointDurationAsync(preFilter, status, metrics.SinceInSeconds(startTime))
+	}()
```

Comment:
why wrapping the function, can't we do

 ```
defer f.metricsRecorder.observeExtensionPointDurationAsync(preFilter, status, metrics.SinceInSeconds(startTime))
```

Reply:
`status` is the return value and at the point of evaluating the deferred function, it's not updated yet.